### PR TITLE
create model and test for Event, only needs to be migrated later

### PR DESCRIPTION
**Pull request by lars3016**

### CODE REVIEW

**Comment by FlorisSp** [comment]
Got rid of a small typo. The rest looks fine.

File: app/models/event.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Event < ApplicationRecord
+  has_one :workshop
```
Comment (arnoFleming) [inline app/models/event.rb]: A workshop has a one to many relation to event, please update accordingly. Also, maybe add code to the other model as well (if you need it!)

File: app/models/event.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  belongs_to :sponsor
+  has_many :attends
```
Comment (arnoFleming) [inline app/models/event.rb]: is `attends` the right model name? We're talking in the tickets about 'attendees'...

File: app/models/event.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+
+  validates :title, presence: true, length: {maximum: 50}
+  validates :locatio, presence: true}
```
Comment (arnoFleming) [inline app/models/event.rb]: syntax error on this line. Do you have tests for this model?

File: app/models/event.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class Event < ApplicationRecord
+  has_one :workshop
+  belongs_to :sponsor
+  has_many :attends
+
+  validates :title, presence: true, length: {maximum: 50}
```
Comment (arnoFleming) [inline app/models/event.rb]: why a max of 50?

File: db/migrate/20170327140451_create_events.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
+class CreateEvents < ActiveRecord::Migration[5.0]
+  def change
+    create_table :events do |t|
+      t.string :title
+      t.string :image
+      t.string :location
+      t.integer :spaces_available
+      t.references :workshop, foreign_key: true
```
Comment (arnoFleming) [inline db/migrate/20170327140451_create_events.rb]: I think this looks OK, but note the comment in the model!

File: spec/models/event_spec.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    it "is invalid without a title" do
+      event = Event.new(title: "")
+      event.valid?
+      expect(event.errors).to have_key(:title)
+    end
+
+    it "is invalid with a description longer than 50 characters" do
+      event = Event.new(description: "A" * 51)
+      event.valid?
+      expect(event.errors).to have_key(:titel)
```
Comment (arnoFleming) [inline spec/models/event_spec.rb]: this test will fail

Comment (arnoFleming) [inline spec/models/event_spec.rb]: can you please show me the results of `bundle exec rspec spec/`?

File: spec/factories/events.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+FactoryGirl.define do
```
Comment (arnoFleming) [inline spec/factories/events.rb]: you don't use this factory. please remove it, or better, use it in your tests

File: spec/models/event_spec.rb [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    end
+
+
+    it "is invalid without a description" do
+      event = Event.new(location: "")
+      event.valid?
+      expect(event.errors).to have_key(:location)
+    end
+  end
```
Comment (arnoFleming) [inline spec/models/event_spec.rb]: please add a spec for a valid event

**Pull request merged**